packages/fast-components/src/text-field/text-field.template.ts

```diff
@@ -0,0 +1,57 @@
+import { html, when } from "@microsoft/fast-element";
+import { ref } from "@microsoft/fast-element/dist/directives";
+import { TextField } from "./text-field";
+
+export const TextFieldTemplate = html<TextField>`
+    ${when(
+        x => x.childNodes.length,
+        html`
+        <label
+            part="label"
+            class="label"
+            for="control"
+        ><slot></slot></label>
+    `
+    )}
+    <div class="root" part="root">
+        <span
+            part="before-content"
+            ${ref("beforeContentContainer")}
+        >
+            <slot
+                name="before-content"
+                @slotchange=${x => x.handleBeforeContentChange()}
+                ${ref("beforeContent")}
+            ></slot>
```

Comment:
I wonder, should the span for before content (and after content) go in their respective slots?

Reply:
If it went into the slot, then user provided slot content would replace that, since content inside a slot is treated as fallback content, rendered only when the user doesn't provide actual content. I this case, I think the span is needed in either case for layout reasons.

Reply:
Correct - we want some basic CSS on the before and after content. Some of which needs to be specific to the left or right of the control. Unfortunately there isn't a methodology for applying the styling to the slot itself via a class so if we want the control to support before and after content without looking broken we need to include the spans to do so